### DISCUSSION TITLE
TRSケーブルで繋いだ方のキーボードが認識されない

### DISCUSSION THREAD

**Discussion by yni77844041** [Yowkees/keyball#245] [Q&A] [answered]
素敵なキーボードキットと丁寧なビルドガイド、ありがとうございます。
Keyball 61 のキットを購入、組み立てました。

本番用のファームウェアを書き込み、全てのキースイッチとLEDの点灯、トラックボールが動作することを確認しています。
USBで直接繋いでいればそれを確認できるのですが、片方をUSBで繋ぎ、TRRSケーブルで繋いでもTRRSでケーブルで繋いだ方のキー、トラックボールが反応しません。LEDは光ります。（たとえば左を繋ぐと右のキーとボールが反応しない、右を繋ぐと左のキーが反応しない）。

![IMG_0875](https://github.com/Yowkees/keyball/assets/28564389/3e18acd6-a647-4427-9bcd-67c08a1b1785)

左手用（トラックボールなし、親指キー3つ）をUSBで繋ぐと、REMAPのサイトで左手用が左右対称に2つ表示されるのですが、関係はありますでしょうか（右を繋いでもこの事象は起こりません）。

<img width="870" alt="スクリーンショット 2023-06-01 16 42 24" src="https://github.com/Yowkees/keyball/assets/28564389/93bae202-eb2d-4222-852e-a1744b476767">

左手用のデバイスのショート、TRSソケットの半田づけは下の通りです。

![IMG_0878](https://github.com/Yowkees/keyball/assets/28564389/5bdd63aa-aeab-4bef-9412-f29eaefcc7d8)

右のTRSソケットは下の通りです。

![IMG_0877](https://github.com/Yowkees/keyball/assets/28564389/7fa3d428-ecda-45d8-a87b-64754f6f18f8)

関係ないかもしれませんが、確認をおこなっているのはMacBook Air (Intel Chip)です。
確認すべきポイントなどご教示いただけますと幸いです。よろしくお願いいたします。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

症状をまとめると、左右のProMicroの通信がうまくいっていないようです。

下記写真、TRSコネクタの半田付けを見直してください。青丸の端子がテンプラしている可能性が高いです。
半田の量が多く見えますので、もう少し落として、半田が基板の銀パッドとなじんで山型になるようにしてください。

もしくは、ごく稀にTRSケーブルが悪い場合もありますので、別のもので試してみてください。

以上お手数ですがご確認お願いいたします。
![61Rfix](https://github.com/Yowkees/keyball/assets/84209772/f6a3a5b3-c493-4cdb-86df-71fa31537582)
![61Lfix](https://github.com/Yowkees/keyball/assets/84209772/d9d9948a-0328-4e5d-aaff-a9f0d2349c85)

**Reply by yni77844041** [reply]
ご確認ありがとうございました。半田を見直したのですが改善せず、TRRSジャックの取り替えを行うつもりでした。
取り外したところ、表側まで流れ込んでいた半田がくっついていたようで、左手側の表側の基板の銀パッドと配線が切れてしまい、配線が一部剥がれかけてしまうという状態になってしまいました。左手側と右手側を交換するケースでない限り関係はなかったでしょうか？ご確認いただけますと幸いです。

![IMG_0880](https://github.com/Yowkees/keyball/assets/28564389/52937552-f3aa-4aaf-8471-467541de349b)

また、ケーブルを新調し、ProMicroのRXIの半田などもこの基板剥がれトラブルの前に確認はしているのですが症状は改善しません。上記の問題が左トラックボールを考慮しなければ関係ないと仮定してのお話ですが、他に見るべきところ（テスターでの導通も含む）がございましたらご教示いただけますと幸いです。よろしくお願いします。

**Comment by Yowkees** [comment]
![KB61TRSfix](https://github.com/Yowkees/keyball/assets/84209772/65dbab74-0412-46ca-948b-6c7964b9ab42)
TRSコネクタ部のパターンが剥がれてしまいましたか。

では、剥がれた部分の端子の半田付けは諦めて、添付画像の青〇のコネクタ端子と、青四角■のパッドをエナメル線やゴム被覆導線で接続してください。これで復活できると思います。
もし基板パッドへの半田付けが難しい場合は■ではなく、線が長くなってしまいますが、青丸●で示した端子と接続してください。

ひとまずこの作業でコネクタを復活させて左右の通信のご確認をお願いします。

**Reply by yni77844041** [reply]
ご確認ありがとうございました！繋がるようになりました。どちらを繋いでもキーを認識するようになりました。
ただ、左右は繋がるようになったのですが、Remapで右側のデバイス（ボールあり）を繋ぐと下記のようになり

<img width="1018" alt="スクリーンショット 2023-06-06 20 25 26" src="https://github.com/Yowkees/keyball/assets/28564389/8cdb76bc-9cea-414b-a9c8-5f99bfe5b929">

左側のデバイス（ボールなし）を繋ぐと下記のようになります。

<img width="1032" alt="スクリーンショット 2023-06-06 20 24 48" src="https://github.com/Yowkees/keyball/assets/28564389/b47106fb-6feb-4e18-9404-251a239de911">

左に繋いでRemapでキーマップを変更しセーブ→右を繋いで同じキーマップの書き込みを行うとRemap上でも正しく認識しますが、つなぎ直したり、ページをリロードすると元の一番上の画像の状態に戻ります。これは正しい状態でしたでしょうか？

また、トラックボールなし側の基盤をダイオードなどの附属品の有無はどちらでも構いませんのでご購入させていただくことはまだ可能でしょうか。動いてはいるのですが不安でして、、、

大変お手数ではございますが、ご確認いただけますと幸いです。よろしくお願いいたします。

**Reply by Yowkees** [reply, marked as answer]
うーんなんででしょう。上画像の状態はおかしいですね。
再度Remapからファームウェアを書き直してどうなるでしょうか？
基板販売の件、了解です。
配送先等のお話をしたいので、お手数ですが下記の白銀ラボショップのお問合せよりメールをいただけますでしょうか。
https://shirogane-lab.com/pages/contact
よろしくお願いいたします。
